### PR TITLE
Ensure initialization of rowIdCounter in AssignUniqueIdOperator

### PR DESCRIPTION
**Pull request by frankobe**
Fix the flakiness of the multithread unit test. 

The root cause is that the values of `rowIdCounter_ `  & `maxRowIdCounterValue_ ` is not initialized in https://github.com/facebookincubator/velox/blob/main/velox/exec/AssignUniqueId.cpp#L80 so the behavior is unpredictable

Signed-off-by: frankobe <frank.hu@bytedance.com>

### CODE REVIEW

File: velox/exec/AssignUniqueId.cpp [hunk 1]
```diff
@@ -44,6 +44,9 @@ AssignUniqueId::AssignUniqueId(
 
   resultProjections_.emplace_back(0, numColumns - 1);
   results_.resize(1);
+
+  // Ensure the rowIdCounter_ is initialized
```
Comment (mbasmanova) [inline velox/exec/AssignUniqueId.cpp:48]: Wouldn't it be sufficient to just initialize rowIdCounter_ and maxRowIdCounterValue_ to zero?

File: velox/exec/AssignUniqueId.cpp [hunk 2]
```diff
@@ -57,38 +60,43 @@ RowVectorPtr AssignUniqueId::getOutput() {
...
   return output;
 }
 
-void AssignUniqueId::generateIdColumn() {
-  auto numInput = input_->size();
+void AssignUniqueId::generateIdColumn(vector_size_t size) {
   auto result = results_[0];
   if (!result || !BaseVector::isReusableFlatVector(result)) {
-    result = BaseVector::create(BIGINT(), numInput, operatorCtx_->pool());
+    result = BaseVector::create(BIGINT(), size, operatorCtx_->pool());
```
Comment (mbasmanova) [inline velox/exec/AssignUniqueId.cpp:72]: This is unrelated change, right?

File: velox/exec/tests/AssignUniqueIdTest.cpp [hunk 3]
```diff
@@ -83,11 +83,13 @@ TEST_F(AssignUniqueIdTest, exceedRequestLimit) {
...
+  for (int i = 0; i < 3; i++) {
+    vector_size_t batchSize = 1000;
+    auto input = {makeRowVector(
+        {makeFlatVector<int32_t>(batchSize, [](auto row) { return row; })})};
 
-  auto plan = PlanBuilder().values(input, true).assignUniqueId().planNode();
+    auto plan = PlanBuilder().values(input, true).assignUniqueId().planNode();
 
-  verifyUniqueId(plan, input, 3);
+    verifyUniqueId(plan, input, 1000);
```
Comment (mbasmanova) [inline velox/exec/tests/AssignUniqueIdTest.cpp:93]: 1'000 threads seems impractical. Why not limit to something more realistic, e.g. 4 or 8?

**Review by mbasmanova — COMMENTED** [review]
@frankobe Thank you for looking into these failures. Some comments and questions below.

File: velox/exec/AssignUniqueId.cpp (hunk 2 above)
Comment (frankobe) [reply velox/exec/AssignUniqueId.cpp:72]: @mbasmanova this is unrelated. Just refactor

File: velox/exec/tests/AssignUniqueIdTest.cpp (hunk 3 above)
Comment (frankobe) [reply velox/exec/tests/AssignUniqueIdTest.cpp:93]: @mbasmanova I increase the value to 1000 to reproduce the issue easier. I can reduce it to 8

File: velox/exec/AssignUniqueId.cpp (hunk 1 above)
Comment (frankobe) [reply velox/exec/AssignUniqueId.cpp]: @mbasmanova U r right, updated

File: velox/exec/tests/AssignUniqueIdTest.cpp [hunk 4]
```diff
@@ -83,11 +83,13 @@ TEST_F(AssignUniqueIdTest, exceedRequestLimit) {
 }
 
 TEST_F(AssignUniqueIdTest, multiThread) {
-  vector_size_t batchSize = 1000;
-  auto input = {makeRowVector(
-      {makeFlatVector<int32_t>(batchSize, [](auto row) { return row; })})};
+  for (int i = 0; i < 3; i++) {
```
Comment (majetideepak) [inline velox/exec/tests/AssignUniqueIdTest.cpp:86]: Is this loop needed?

File: velox/exec/AssignUniqueId.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/exec/AssignUniqueId.cpp:72]: It would be easier to review if a PR was limited to just a fix or just a refactor. Otherwise, it is hard to separate out material changes.

File: velox/exec/AssignUniqueId.cpp [hunk 5]
```diff
@@ -57,38 +60,43 @@ RowVectorPtr AssignUniqueId::getOutput() {
...
+    auto end = std::min(size, (int32_t)(start + kRowIdsPerRequest));
 
     std::iota(
         rawResults + start, rawResults + end, uniqueValueMask_ | rowIdCounter_);
     rowIdCounter_ += end;
     start = end;
   }
 }
+
+void AssignUniqueId::requestRowId() {
```
Comment (mbasmanova) [inline velox/exec/AssignUniqueId.cpp:95]: requestRowId - >requestRowIds since this method requests a batch of IDs, not just one.

File: velox/exec/AssignUniqueId.cpp [hunk 6]
```diff
@@ -57,38 +60,43 @@ RowVectorPtr AssignUniqueId::getOutput() {
...
+  rowIdCounter_ = rowIdPool_->fetch_add(kRowIdsPerRequest);
+  maxRowIdCounterValue_ =
+      std::min({rowIdCounter_ + kRowIdsPerRequest, kMaxRowId});
+  VELOX_CHECK_LT(
```
Comment (mbasmanova) [inline velox/exec/AssignUniqueId.cpp:99]: This is a fix for another issue, right? E.g. the loop above could got on forever without that fix. Is there a way to add test coverage for this case?

File: velox/exec/AssignUniqueId.cpp (hunk 2 above)
Comment (frankobe) [reply velox/exec/AssignUniqueId.cpp:72]: @mbasmanova I will break this PR into 2

File: velox/exec/tests/AssignUniqueIdTest.cpp (hunk 4 above)
Comment (frankobe) [reply velox/exec/tests/AssignUniqueIdTest.cpp:86]: @majetideepak I will prefer to keep this loop so the unit test has higher chance to fail, instead of flakiness

File: velox/exec/AssignUniqueId.cpp (hunk 6 above)
Comment (frankobe) [reply velox/exec/AssignUniqueId.cpp:99]: @mbasmanova This is added to prevent rowIdCounter_ goes above 40 bit which will take 1 << 20L batches to reach. I can add a unit test in a separate PR. Not related with the current bug

**Review by mbasmanova — APPROVED** [review]
Thank you, @frankobe .
